Source/Private/Invoke-Command.ps1

```diff
@@ -67,28 +68,39 @@ function Invoke-Command {
         while (-not $process.WaitForExit(100)) {
             # Allow interrupts like CTRL + C by doing a non-blocking wait
         }
-        # Allow all async event handlers to finish up by doing a blocking wait
+        # Allow all async event handlers to finish up by doing a blocking wait, then free up the resources
+        # associated with that process on OS level.
         $process.WaitForExit()
+        $processId = $process.Id
+        $result.ExitCode = $process.ExitCode
+        $process.Close()
```

Comment:
If the process is closed here, not sure how the finally block will `kill()` the closed process.
Can this be tested manually? Automated testing for this scenario is rather impossible.

Reply:
In case the call to Close fails, the Kill call is the best we can do. Calling Close() will request the process to stop and release OS resources. Kill is more like pulling the plug which may leave the OS in a state where the clean was not done correctly. 

The call to Close() should not throw exceptions but prior code might.

I am not sure how we could test this. I guess it would require some pretty low-level hooks for manipulating processes at the OS level. Don't know where to start 😅